frontend/src/routes/UserManagement/Roles/Role/RolePermissions.tsx

```diff
@@ -1,18 +1,111 @@
 /* Copyright Contributors to the Open Cluster Management project */
-import { PageSection } from '@patternfly/react-core'
-import { useParams } from 'react-router-dom-v5-compat'
+import { Flex, FlexItem, Label, PageSection } from '@patternfly/react-core'
+
 import { useTranslation } from '../../../../lib/acm-i18next'
-import { AcmPage, AcmPageContent, AcmPageHeader } from '../../../../ui-components'
+import { AcmPage, AcmPageHeader, AcmTable, compareStrings, IAcmTableColumn } from '../../../../ui-components'
+
+import { PolicyRule } from '../../../../resources/kubernetes-client'
+import { useMemo, useCallback } from 'react'
+import { useCurrentRole } from '../RolesPage'
+import { cellWidth } from '@patternfly/react-table'
+
+export const kindToAbbreviation = (resourceOrKind: string): string => {
+  // Step 1: Extract the resource name part (before any path separators)
+  const resourceName = resourceOrKind.split(/[/\\?#]/)[0] || resourceOrKind
+
+  // Complex words: virtualmachines → VIRT, persistentvolumes → PERS
+  const abbrKind = resourceName.toUpperCase().slice(0, 4)
+  // Using logic in openshift console code
+  // https://github.com/openshift/console/blob/a228536605a2451468d172da9bb7b6ab812511ea/frontend/public/module/k8s/get-resources.ts#L29
+  const blocklist = ['ASS', 'FART']
+  return blocklist.includes(abbrKind) ? abbrKind.slice(0, -1) : abbrKind
+}
 
 const RolePermissions = () => {
   const { t } = useTranslation()
-  const { id = undefined } = useParams()
+  const currentRole = useCurrentRole()
+
+  const policyRules = useMemo(() => {
+    if (!currentRole?.rules) return []
+    return currentRole.rules
+  }, [currentRole])
+
+  const columns = useMemo<IAcmTableColumn<PolicyRule>[]>(
+    () => [
+      {
+        header: t('Actions'),
+        sort: (a: PolicyRule, b: PolicyRule) => compareStrings((a.verbs || []).join(', '), (b.verbs || []).join(', ')),
+        search: (item: PolicyRule) => (item.verbs || []).join(' '),
+        cell: (item) => {
+          return (
+            <div>
+              {(item.verbs || []).map((verb, index) => (
+                <div key={index}>
+                  <strong>{verb}</strong>
+                </div>
+              ))}
+            </div>
+          )
+        },
```

Comment:
in general if a function is just returning something brackets can be avoided to reduce boilerplate. It is something you are already doing on many places (even on this file), so it is just about to keep consistency)

```suggestion
        cell: (item) => (
            <div>
              {(item.verbs || []).map((verb, index) => (
                <div key={index}>
                  <strong>{verb}</strong>
                </div>
              ))}
            </div>
          ),
```